File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int start = 0, end = 0, answer = INT_MAX;  
        while(end < nums.size()){
            target -= nums[end]; 

            while(target <= 0){
                answer = min(answer, end - start + 1); 
                target += nums[start++]; 
            }

            end++; 
        }

        return answer == INT_MAX ? 0 : answer; 
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int start = 0, end = 0; 
        int answer = Integer.MAX_VALUE; 
        int sum = 0; 
        while(end < nums.length){
            sum += nums[end]; 
            while(sum >= target){
                answer = Math.min(answer, end - start + 1); 
                sum -= nums[start++]; 
            }
            end++; 
        }
        return answer == Integer.MAX_VALUE ? 0 : answer; 
    }
}